src/computePrizeDistributionFromTicketAverageTotalSupplies.ts

```diff
@@ -0,0 +1,90 @@
+import { BigNumber, BigNumberish } from "@ethersproject/bignumber";
+import { Draw, PrizeDistribution, PrizeTier } from "./types";
+import {
+  calculatePicksFromAverageTotalSuppliesBetween,
+  computeCardinality,
+} from "./index";
+import { sumBigNumbers } from "./utils";
+const debug = require("debug")("v4-js-core");
+
+function createBigNumber(value: BigNumberish) {
+  return BigNumber.from(value);
+}
+
+export async function computePrizeDistributionFromTicketAverageTotalSupplies(
+  draw: Draw,
+  prizeTier?: PrizeTier,
+  ticketPrimaryAverageTotalSupply?: BigNumberish,
+  ticketSecondaryListAverageTotalSupply?: Array<BigNumberish>,
+  decimals: BigNumberish = 18
+): Promise<PrizeDistribution | undefined> {
+  if (
+    !draw ||
+    !prizeTier ||
+    !ticketPrimaryAverageTotalSupply ||
+    !ticketSecondaryListAverageTotalSupply
+  )
+    return undefined;
```

Comment:
If we need these values, they should just be required in the function params. Would avoid using all these `undefined`.